app/src/components/MemberBox.jsx

```diff
@@ -0,0 +1,235 @@
+import React, { Component } from 'react';
+
+import styled from 'styled-components';
+
+// STYLED COMPONENTS
+const MemerBoxContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  width: 250px;
+`;
+
+const MemberList = styled.div`
+  display: flex;
+  flex-direction: column;
+  float: left;
+  width: 250px;
+  overflow-y: auto;
+  height: 200px;
+  margin: 10px 0;
+`;
+
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const Button = styled.button`
+  background: none;
+  color: inherit;
+  border: none;
+  padding: 0;
+  font: inherit;
+  cursor: pointer;
+  outline: inherit;
+  color: blue; 
+  text-decoration: underline;
+
+  &:hover{
+    color: darkblue;
+  }
+`;
+
+const Input = styled.input`
+  padding: 5px 8px;
+`;
+
+// INLINE COMPONENTS (ONLY FOR TOGGLING)
+const addMemberForm = {
+  flexDirection: "column",
+  marginTop: "20px",
+  display: "none",
+  height: "120px",
+  border: "1px solid lightgray",
+  justifyContent: "space-between",
+  padding: "10px 20px",
+  background: "whitesmoke"
+}
+
+const formError = {
+  color: 'red',
+  fontSize: '0.9rem',
+  display: 'none',
+}
+
+// TEMPORARY CONSTANT (REPLACED BY AJAX REQUEST)
+const groupMembers = [
+  {name: 'Frodo Baggins', phone: 123444, isChecked: false}, 
+  {name: 'Peregrin  Took', phone: 1234, isChecked: false}, 
+  {name: 'Meriadoc Brandybuck', phone: 12345, isChecked: false}, 
+  {name: 'Famwise Gamgee', phone: 123456, isChecked: false},
+  {name: 'Wrodo Baggins', phone: 1258883, isChecked: false}, 
+  {name: 'Reregrin  Took', phone: 825485, isChecked: false}, 
+  {name: 'Seriadoc Brandybuck', phone: 66672, isChecked: false}, 
+  {name: 'Ramwise Gamgee', phone: 235235, isChecked: false}
+]
+
+class MemberBox extends Component {
+  constructor() {
+    super();
+    this.state = {
+      members: [],
+      name: '',
+      phone: '',
+    }
+  }
+
+  componentDidMount() {
+    // GET request goes here:
+    this.setState({
+      members: groupMembers,
+    })
+  }
+
+  toggleCheckbox = (phone) => {
+    let updatedMembers = [...this.state.members];
+
+    updatedMembers.map(member => {
+      (member.phone === phone &&
+        (member.isChecked = !member.isChecked) 
+      )
+    })
+
+    this.setState({
+      members : updatedMembers,
+    })
+  }
+
+  toggleMemberForm = () => {
+    let memberForm = document.querySelector('.addMemberForm');
```

Comment:
it's fairly rare that you want to use querySelector in react. usually you want to update your data, and then have your data get referenced in the render. so you would have memberFormToggle on state, and then in the render you would do:
```
<Component style={{ display: this.state.memberFormToggle ? 'flex' : 'none` }} />
```
and then in your method you would so something like:
```
toggleMemberForm = () => {
    this.setState(prevState => ({ memberFormToggle: !prevState.memberFormToggle })
}
```

Reply:
I love these comments, thank you. I implemented this and its functional. Super clean. Though I did make a minor change to toggleMemberForm:

```
  toggleMemberForm = () => {
    this.setState(prevState => ({ 
      memberFormToggle: !prevState.memberFormToggle
    }) );
  }
```